src/fastga_he/powertrain_builder/unit_tests/test_power_train_file.py

```diff
@@ -178,6 +176,48 @@ def test_power_train_file_connections():
         print("[" + om_output + ", " + om_input + "]")
 
 
+def test_power_train_initialization_time():
+    sample_power_train_file_path = pth.join(pth.dirname(__file__), "data", YML_FILE)
+    power_train_configurator = FASTGAHEPowerTrainConfigurator(
+        power_train_file_path=sample_power_train_file_path
+    )
+
+    power_train_configurator._get_components()
+    power_train_configurator._get_connections()
+
+    cache = power_train_configurator._cache[sample_power_train_file_path]
+
+    print("\n")
+    print("Simple assembly")
+    print(f"Load time: {cache['load_time']} sec")
+    print(f"Get component time: {cache['get_component_time']} sec")
+    print(f"Get connection time: {cache['get_connection_time']} sec")
+    print("=" * 50)
+    sample_power_train_file_path = pth.join(pth.dirname(__file__), "data", "octo_assembly.yml")
+    power_train_configurator = FASTGAHEPowerTrainConfigurator(
+        power_train_file_path=sample_power_train_file_path
+    )
+
+    power_train_configurator._get_components()
+    power_train_configurator._get_connections()
+
+    file_path = power_train_configurator._cache.keys()
+
+    for path in file_path:
```

Comment:
Why iterate when we have already looked at the sample one so there is only the octo_assembly remaining ?